elastic/kibana
# [Security Solutions] Add PLI authorisation for Investigation Guide

## Summary

Add PLI authorization checks and Upselling message to Investigation.
*This PR restricts access to the features* and creates an updated upselling hover message.
* It updates the Upselling registering to accept string because the Markdown component doesn't accept components as hover messages.

### How to test it?
* Open timeline
* Go to the notes tab
* You should find an investigation guide button on the markdown editor toolbar


#### ESS `yarn start`
* Run ESS with a basic license
  * It should not change
* Run ESS with a platinum
  * It should not change
  
#### Serverless `yarn serverless-security`
* Run Serverless with security essentials (serverless.security.yml)
   * It should show the Upselling message
```
xpack.serverless.security.productTypes:
  [
    { product_line: 'security', product_tier: 'essentials' }
  ]
```
* Run Serverless with security complete (kibana/config/serverless.security.yml)
  * It should show the Investigation guide button
```
xpack.serverless.security.productTypes:
  [
    { product_line: 'security', product_tier: 'complete' },
  ]
 
 ```

<img width="1761" alt="Screenshot 2023-07-31 at 09 42 41" src="https://github.com/elastic/kibana/assets/1490444/1a1e0313-7335-4a20-84e3-ec2d48f80b9c">



### Checklist

Delete any items that are not applicable to this PR.
